game/operation/operation.py

```diff
@@ -503,10 +506,33 @@ def generate_lua(
                         },
                     }
 
+        for cp in cls.game.theater.controlpoints:
+            for ground_object in cp.ground_objects:
+                if ground_object.might_have_aa and not ground_object.is_dead:
+                    for g in ground_object.groups:
+                        threat_range = ground_object.threat_range(g)
+
+                        if not threat_range or threat_range.distance_in_meters == 0.0:
```

Comment:
if not threat_range.meters:

`distance_in_meters` is the "private" implementation detail, and `not threat_range` currently does nothing for a non-`Optional[Distance]` . If you wanted to define `__bool__` for `Distance` you could do `if not threat_range`.

Reply:
Done

Reply:
Does this work they way you have it now? Do you not need to add `def __bool__` to `Distance`? aiui this will always be `True`